common/djangoapps/entitlements/models.py

```diff
@@ -307,26 +311,33 @@ def get_active_entitlements_for_user(cls, user):
         ).select_related('user').select_related('enrollment_course_run')
 
     @classmethod
-    def does_user_have_active_entitlements(cls, user):
+    def get_fullfillable_entitlements(cls, user):
         """
-        Checks whether or not a user has any active entitlements
+        Returns the most recent Course Entitlement for a Course that a user can use
+
+        Arguments:
+            user (User): The user we are looking at the entitlements of.
+            course_uuid: The Course UUID of the course we are looking for.
+
+        Returns
+            list<CourseEntitlements>: A list of course Entitlements that a user can enroll in. These must not be expired
+                and not have a course run already assigned to it.
         """
-        count = cls.objects.count(
-            user=user
-        ).exclude(
-            expired_at__isnull=False,
-            enrollment_course_run=None
-        ).select_related('user').select_related('enrollment_course_run')
-        return count > 0
+
+        try:
+            return cls.objects.filter(
+                user=user,
+            ).exclude(
+                expired_at__isnull=False,
+                enrollment_course_run__isnull=False
+            ).order_by('created')
+        except CourseEntitlement.DoesNotExist:
+            return []
 
     @classmethod
     def get_fullfillable_entitlement_for_user(cls, user, course_run_key):
         """
-        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the
-        course run provided in the course_key.  If the user does have an Entitlement with no run set, the User is
-        enrolled in the mode set in the Entitlement.
-
-        Note: This method assumes that eligibility to enroll in the course has already been checked.
+        Retrieves a fullfillable entitlement for the user and the Course given.
```

Comment:
For clarity, I would explain in this high level part of the docstring that you are actually using course run (rather than the more general course) to find the user's fulfillable entitlement.